lib/go/edgecontext/edgecontext.go

```diff
@@ -24,6 +25,13 @@ const LoIDPrefix = "t2_"
 // does not have the correct prefix.
 var ErrLoIDWrongPrefix = errors.New("edgecontext: loid should have " + LoIDPrefix + " prefix")
 
+// Locale codes should contain either a language, or a language and region specifier
+// separated by an underscore.
+// e.g. en, en_US
+const LocaleRegex = "^[a-z]{2}(_[A-Z]{2})?$"
```

Comment:
```suggestion
var LocaleRegex = regexp.MustCompile(`^[a-z]{2}(_[A-Z]{2})?$`)
```

this way we only compile the regexp once (currently every `regexp.MatchString` call would compile it again).

Reply:
and also please add a few unit tests to make sure that this regexp matches `en`, `en_US` and not `en-US` (and maybe a few other cases).

Reply:
Updated. And tests added in https://github.com/reddit/edgecontext/pull/12/commits/3f3511801f0b168d0f4de6e8fb53252d917d4170